api/src/handlers.rs

```diff
@@ -118,18 +118,35 @@ impl OutputHandler {
 		include_proof: bool,
 	) -> BlockOutputs {
 		let header = w(&self.chain).get_header_by_height(block_height).unwrap();
-		let block = w(&self.chain).get_block(&header.hash()).unwrap();
-		let outputs = block
-			.outputs
-			.iter()
-			.filter(|output| commitments.is_empty() || commitments.contains(&output.commit))
-			.map(|output| {
-				OutputPrintable::from_output(output, w(&self.chain), &block, include_proof)
-			})
-			.collect();
-		BlockOutputs {
-			header: BlockHeaderInfo::from_header(&header),
-			outputs: outputs,
+
+		// TODO - possible to compact away blocks we care about
```

Comment:
If we compact blocks from the index before we have refreshed a coinbase output then we lose the ability to spend it.
But wallet code is handling this correctly now and does not blow up.
We just don't select these outputs as spendable.

To tackle in a separate PR - retrieve block and merkle proof as soon as we mine a block (don't wait to manually refresh the wallet).